lib/api/apiUtils/object/createAndStoreObject.js

```diff
@@ -193,6 +193,10 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
         metadataStoreParams.oldReplayId = objMD.uploadId;
     }
 
+    if (objMD && !bucketMD.isVersioningEnabled() && objMD.archive && Object.keys(objMD.archive).length > 0) {
```

Comment:
* This handles the case of bucket without versioning ; we may however have the same issue in version suspended case ? @Kerkesni, is there maybe already a function to check if we are in these kind of condition?
* no reason to check Object.keys(objMD.archive) : we should simply check if the object has been written to "cold", i.e. `objMD?.archive?.archiveInfo` (and you must not check the content of `archiveInfo`, it is supposed to stay opaque: hence the type `any` in arsenal's `ObjectMDArchive`)

AFAIR, race conditions (e.g. object deleted/replaced/...) during the archival process should be handled already when finishing the transition : if the object does not exist (or etag does not match), backbeat will post a Delete/GC request back to tape I think...

Reply:
Looking at the function  `isVersioningEnabled`
```
    isVersioningEnabled() {
        const versioningConfig = this.getVersioningConfiguration();
        return versioningConfig ? versioningConfig.Status === 'Enabled' : false;
    }
```
This should handle the versioning suspended case also.